File: uri_1024_encryption.c
#include <stdio.h>
#include <string.h>

int main(void)
{
	int cases; char word[1000]; char reverse[1000];
	scanf("%d", &cases);
	int i;
	for(i = 0; i < cases; i++){
		scanf(" %[^\n]", word);
		int c;
		for(c = 0; c < strlen(word); c++){
			if(word[c] >= 65 && word[c] <= 90 || word[c] >= 97 && word[c] <= 122){
				word[c] += 3;
			}
		}
        int size_word = strlen(word) - 1;
		c = 0;
        while(size_word >= 0){
			reverse[c] = word[size_word];
			size_word--;
            c++;
		}
        reverse[c] ='\0';
        for(c = 0; c < strlen(word); c++){
            if(c >= strlen(word)/2){
                reverse[c]--;
            }
            printf("%c", reverse[c]);
        }
        printf("\n");
	}
	return 0;
}

File: uri_1024_encryption.py
from math import trunc
cases = int(input())
for i in range(cases):
    word = input()
    reverse_word = []
    for c in range(len(word)):
        reverse_word.append(word[c])
        if ord(word[c]) >= 65 and ord(word[c]) <= 90 or ord(word[c]) >= 97 and ord(word[c]) <= 122:
            reverse_word[c] = chr(ord(word[c]) + 3)
    
    reverse_word.reverse()
    c = trunc(len(reverse_word)/2)
    while c < len(reverse_word):
        reverse_word[c] = chr(ord(reverse_word[c]) - 1)
        c += 1
    
    print(*reverse_word, sep="")